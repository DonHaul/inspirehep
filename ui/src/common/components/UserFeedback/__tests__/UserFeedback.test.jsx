import { render } from '@testing-library/react';
import { Modal, Rate, Input } from 'antd';

import { shallow } from 'enzyme';
import { trackEvent, checkIsTrackerBlocked } from '../../../../tracker';
import UserFeedback from '../UserFeedback';

jest.mock('../../../../tracker');

describe('UserFeedback', () => {
  afterEach(() => {
    trackEvent.mockClear();
    checkIsTrackerBlocked.mockClear();
  });

  it('renders when tracker is blocked', () => {
    checkIsTrackerBlocked.mockImplementation(() => true);
    const { getByText, getByRole } = render(<UserFeedback />);

    getByRole('button').click();
    expect(getByText('AdBlock detected')).toBeInTheDocument();
  });

  it('sets modal visible true on feedback button click', () => {
    const { getByTestId, getByRole } = render(<UserFeedback />);
    getByRole('button').click();
    expect(getByTestId('user-feedback')).toBeInTheDocument();
  });

  it('calls trackEvent with feedback on modal Ok and renders thank you', () => {
    const rateValue = 3;
    const commentValue = 'Not bad';
    const wrapper = shallow(<UserFeedback />);
    wrapper.find(Input.TextArea).prop('onChange')({
      target: { value: commentValue },
    });
    wrapper.find(Rate).prop('onChange')(rateValue);
    const onModalOk = wrapper.find(Modal).prop('onOk');
    onModalOk();
    expect(trackEvent).toHaveBeenCalledWith(
      'Feedback modal',
      'Feedback submission',
      `Feedback comment: ${commentValue}`,
      rateValue
    );
    expect(wrapper).toMatchSnapshot();
  });
});
