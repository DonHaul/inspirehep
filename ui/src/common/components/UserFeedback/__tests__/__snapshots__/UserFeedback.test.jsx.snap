// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFeedback calls trackEvent with feedback on modal Ok and renders thank you 1`] = `
<div
  className="__UserFeedback__"
>
  <Button
    className="feedback-button"
    data-test-id="sticky"
    icon={<MessageOutlined />}
    onClick={[Function]}
    size="large"
    type="primary"
  >
    <ResponsiveView
      min="sm"
      render={[Function]}
    />
  </Button>
  <Modal
    afterClose={[Function]}
    data-testid="user-feedback"
    footer={null}
    okButtonProps={
      Object {
        "disabled": true,
      }
    }
    okText="Submit"
    onCancel={[Function]}
    onOk={[Function]}
    open={false}
    title="Your Feedback"
  >
    <ModalSuccessResult>
      <div>
        Thank you for your response.
      </div>
      <div>
        For further feedback, please
         
        <LinkWithTargetBlank
          as="a"
          href="https://forms.gle/ZQi31GvXXHcsgXgM6"
        >
          take our survey
        </LinkWithTargetBlank>
        .
      </div>
      <div>
        It takes around 5 minutes to complete.
      </div>
    </ModalSuccessResult>
  </Modal>
</div>
`;
